### ISSUE TITLE
Error: Invalid isoformat string

### ISSUE THREAD

**Issue by mrjones-plip** [NegativeK/picamserver#24]
I think I loaded everything correctly?  But I was testing rappid reloads of `/printer` endpoint when the camera server crashed with this error (below).  It have been a fluke or  PEBCAK, but filing in case we can puzzle out what happened and defend against it. 

```
[65:35:23.258565001] [2581326]  INFO Camera camera.cpp:1197 configuring streams: (0) 4056x3040-BGR888 (1) 4056x3040-SBGGR12_CSI2P             15:15:30 [46/83]
[65:35:23.263143975] [2581318]  INFO RPI vc4.cpp:622 Sensor: /base/soc/i2c0mux/i2c@1/imx477@1a - Selected sensor format: 4056x3040-SBGGR12_1X12 - Selected uni
cam format: 4056x3040-pBCC                                                                                                                                    
Removing expired session file.                                                                                                                                
Removing expired session file.                                                                                                                                
Traceback (most recent call last):                                                                                                                            
  File "/root/picamserver/camdaemon.py", line 74, in <module>                                                                                                 
    main()                                                                                                                                                    
  File "/root/picamserver/camdaemon.py", line 61, in main                                                                                                     
    run_camera_loop(picam2)                                                                                                                                   
  File "/root/picamserver/camdaemon.py", line 50, in run_camera_loop                                                                                          
    if web_listening():                                                                                                                                       
       ^^^^^^^^^^^^^^^                                                                                                                                        
  File "/root/picamserver/camdaemon.py", line 37, in web_listening             
    session_timestamp = datetime.datetime.fromisoformat(session_contents)                                                                                     
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                     
ValueError: Invalid isoformat string: ''                
```

**Comment by NegativeK** [comment]
Weird! Appreciate the stack trace. Any thoughts on how you were rapid reloading? And did it reproduce ever?

**Comment by mrjones-plip** [comment]
I was simply mashing "refresh" in my browser and watching the back end.  I was unable to reproduce :(

**Comment by NegativeK** [comment]
A way to (sometimes) reproduce:

1. Ensure camdaemon.py is running.
2. Close all clients that are loading the image endpoint.
3. Ensure the listeners directory has no files in it. camdaemon.py will do this after 30 seconds.
4. Run: `for i in {1..100}; do curl -s 127.0.0.1:5000/printer -O /dev/null & done`.

It's somewhat abusive to the local system. Upping the number from 100 seems to decrease reproducibility.

**Comment by mrjones-plip** [comment]
nice sleuthing!

minor point: `-O` will use the file name in the URL, `printer` in this case, while `-o` will use the path specified, `/dev/null`.

**Comment by mrjones-plip** [comment]
oh  - on more thing - this does NOT work per our IRL conversation:

`watch -n 0.01 "curl -s 127.0.0.1:5000/printer -o /dev/null &"`

I think A) `watch` only goes to `0.1` and B) I think it's waiting for the returned response

**Comment by NegativeK** [comment]
Was resolved in PR #33 .
